sirepo/auth/__init__.py

```diff
@@ -669,12 +681,14 @@ def _parse_display_name(value):
     return res
 
 
-def _plan(roles):
-    if 'enterprise' in roles:
-        return 'enterprise'
-    elif 'premium' in roles:
-        return 'premium'
-    return 'basic'
+def _plan(data):
+    r = data.roles
+    data.upgradeToPlan = 'enterprise' if 'premium' in r else 'premium'
```

Comment:
This should be None in the case of `enterprise` so:

```py
    if ROLE_PAYMENT_PLAN_ENTERPRISE in r:
        data.paymentPlan = _PAYMENT_PLAN_ENTERPRISE
        data.updateToPlan = None
    elif ROLE_PAYMENT_PLAN_PREMIUM in r:
        data.paymentPlan = _PAYMENT_PLAN_PREMIUM
        data.upgradeToPlan = _PAYMENT_PLAN_ENTERPRISE
   else:
        data.paymentPlan = _PAYMENT_PLAN_BASIC
        data.upgradeToPlan = _PAYMENT_PLAN_ENTERPRISE
```

Reply:
https://github.com/radiasoft/sirepo/pull/2711/commits/05bde41c8e65d6cbc52101463c4a1a6418c2356b